src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Dynamo.Core;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.Nodes;
+using Dynamo.Wpf.Authentication;
+using Greg;
+using Greg.AuthProviders;
+using Reach;
+using Reach.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Dynamo.Publish.Models
+{
+    public class PublishModel
+    {
+        private readonly IAuthProvider authenticationProvider;
+        private readonly ICustomNodeManager customNodeManager;
+
+        private readonly string serverUrl;
+        private readonly string port;
+        private readonly string page;
+
+        private IWorkspaceStorageClient reachClient;
+
+        public bool IsLoggedIn
+        {
+            get
+            {
+                return authenticationProvider.LoginState == LoginState.LoggedIn;
+            }
+        }
+
+        public HomeWorkspaceModel HomeWorkspace
+        {
+            get;
+            private set;
+        }
+
+        public List<CustomNodeWorkspaceModel> CustomNodesWorkspaces
+        {
+            get;
+            private set;
+        }
+
+        #region Initialization
+
+        public PublishModel(IAuthProvider dynamoAuthenticationProvider, ICustomNodeManager dynamoCustomNodeManager)
```

Comment:
You can use the `InternalsVisibleToAttribute` to get around this. You'll need to add this to the assembly info.